src/service/discovery.js

```diff
@@ -180,14 +183,17 @@ const fetchInitialDiscoveryTopics = async (limit = 50, page = 0) => {
  * @param {Number} page
  * @returns {FetchInitialDiscoveryUsersResponse}
  */
-const fetchInitialDiscoveryUsers = async (limit = 50, page = 0) => {
+const fetchInitialDiscoveryUsers = async (limit = 50, page = 0, isAnon = false) => {
   const body = {
     page
   };
 
   if (limit) body.limit = limit;
   try {
-    const response = await api.post(`discovery/init/user?limit=${limit}&page=${page}`, body);
+    const response = await api.post(
+      `discovery/init/user?limit=${limit}&page=${page}${isAnon ? '&allow_anon_dm=true' : ''}`,
+      body
+    );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `limit` parameter is being added to the request body and also appended as a query parameter in the URL. This seems redundant. If the API endpoint accepts it as a query parameter, there's no need to include it in the request body. The same applies to the `page` parameter.
```diff
-    const body = {
-      page
-    };
-
-    if (limit) body.limit = limit;
+    const body = {};
```
```diff
-       `discovery/init/user?limit=${limit}&page=${page}${isAnon ? '&allow_anon_dm=true' : ''}`,
+       `discovery/init/user?page=${page}${isAnon ? '&allow_anon_dm=true' : ''}`,
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->